controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -204,6 +214,12 @@ public void initializeStreamWriters(final ClientFactory clientFactory,
                                 Throwable cause = ExceptionHelpers.getRealException(e);
                                 if (cause instanceof ScaleOperationExceptions.ScalePreConditionFailureException) {
                                     response.setStatus(ScaleResponse.ScaleStreamStatus.PRECONDITION_FAILED);
+                                } else if (cause instanceof ScaleOperationExceptions.ManualScaleTimedOut) {
+                                    // Note: timedout does not mean the scale operation failed. In fact scale operation once started,
```

Comment:
Timeout here seems like a workaround to ensure clients do not wait for too long and does not have any real meaning to the operation itself. Can we split the API into the following 2 instead (making in an async API) :
1. triggerManualScale - API returns as soon as manualScale is triggered (with same responses as before) and does not block
2. checkScaleStatus - This can be called by client any number of times and indicates if scale completed, responses could be an enum with status  - Completed, Failed, InProgress, NotFound. Now the clients can decide how long to wait with no extra overhead on the server.